src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -2080,6 +2080,18 @@ void G1ConcurrentMark::print_on_error(outputStream* st) const {
   _mark_bitmap.print_on_error(st, " Bits: ");
 }
 
+void G1ConcurrentMark::update_concurrent_mark_threads_cpu_time() {
+  assert(Thread::current() == cm_thread(),
+         "Must be called from _cmThread to avoid races");
+  if (!UsePerfData || !os::is_thread_cpu_time_supported()) {
+    return;
+  }
+  CPUTimeCounters* counters = _g1h->cpu_time_counters();
+  ThreadTotalCPUTimeClosure tttc(counters, CPUTimeGroups::gc_conc_mark);
+  tttc.do_thread(cm_thread());
+  threads_do(&tttc);
+}
```

Comment:
Any particular reason for having this in `G1ConcurrentMark` instead of keeping it where it is called in `G1ConcurrentMarkThread`. The implementation would be more or less the same apart from the two last lines:
```
  tttc.do_thread(this);
  cm()->threads_do(&tttc);
```